### PR TITLE
[BUG fix] solve the CI bug of router fusion

### PR DESCRIPTION
**Pull request by Autumn1998**
# Description

Fixes for Router Fusion:
1. Corrected kernel dispatching issue on SM90 architecture.
2. Resolved CI bug by using properly constructed input data in unit tests.
3. Fixed bug occurring when MBS > 1.

Fixes # (issue)

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes

Please list the changes introduced in this PR:

- Change A
- Change B

# Checklist:

- [ ] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [ ] The functionality is complete
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/pytorch/test_fused_router.py [hunk 1]
```diff
@@ -148,11 +148,15 @@ def run_comparison(
 ):
     # Set some parameters
     if score_function == "sigmoid":
-        logits = torch.arange(num_experts, device="cuda", dtype=dtype) * 0.1
+        # Construct the special logits to avoid inf in the sigmoid function
+        logits = 1 + torch.arange(num_experts, device="cuda", dtype=dtype) * 0.01
         logits = logits.unsqueeze(0).repeat(num_tokens, 1)
+        random_values = torch.rand(num_tokens, num_experts, device="cuda")
+        _, indices = torch.sort(random_values, dim=1)
+        logits = torch.gather(logits, 1, indices)
```
Comment (timmoon10) [inline tests/pytorch/test_fused_router.py]: Can you explain how this helps avoid infs? Was the problem that `logits` became too large, causing sigmoid to output 1.0 in multiple entries? Also, I don't understand the benefit from forcing `logits >= 1`. Previously we had `logits >= 0` and `sigmoid(0) = 0.5`, so I wouldn't expect us to have had divide-by-zero problems.

**Review by timmoon10 — APPROVED** [review]
Overall LGTM, thanks for the fix!

File: tests/pytorch/test_fused_router.py (hunk 1 above)
Comment (Autumn1998) [reply tests/pytorch/test_fused_router.py]: @timmoon10  There are some changes in this part.
In summary, we need to consider two main issues:

1. When using `top-k` on `M` values, the `k-th` and `(k+1)-th` value can sometimes be the same. This may cause fused implementation and PyTorch’s implementation to produce different results. To fix this, we need to ensure that all input values in each row are unique. (Note: using the random API is unreliable even if the seed is fixed.)
2. The sigmoid function can easily produce `inf` if the input is too large, due to the exp(-x)

To fix these, I created a special input matrix that satisfies both conditions and reduced the problem size. There is no need to enforce `logits >=1`; I only used that constraint in the initial version to find a suitable value range to avoid the inf.

**Review by yaox12 — APPROVED** [review]
